1_fetch/src/fetch_precip_spatial.R

```diff
@@ -0,0 +1,51 @@
+
+#' @param ind_file scipiper indicator file
+#' @param view_config with `projection`, a valid input to sf::st_crs
+#' and `bbox` e.g. [-83, 25, -80, 28]
+fetch_precip_spatial <- function(ind_file, view_config) {
+
+  nc <- ncdf4::nc_open("https://cida.usgs.gov/thredds/dodsC/stageiv_combined")
+
+  lon <- ncdf4::ncvar_get(nc, "lon")
+  lat <- ncdf4::ncvar_get(nc, "lat")
+
+  x <- matrix(rep(c(1:nrow(lon)), ncol(lon)),
+              nrow = nrow(lon), ncol = ncol(lon),
+              byrow = FALSE)
+
+  y <- matrix(rep(c(1:ncol(lon)), nrow(lon)),
+              nrow = nrow(lon), ncol = ncol(lon),
+              byrow = TRUE)
+
+  sf_points <- data.frame(x = matrix(x, ncol = 1),
+                          y = matrix(y, ncol = 1),
+                          lon = matrix(lon, ncol = 1),
+                          lat = matrix(lat, ncol = 1))
+
+  sf_points <- sf::st_as_sf(sf_points,
+                            coords = c("lon", "lat"),
+                            crs = "+init=epsg:4326",
+                            agr = "constant")
+
+  bbox <- bbox_to_polygon(bbox = view_config$bbox,
```

Comment:
This looks good, but I think you should use `view_polygon` as that gives the whole visible domain, vs the initial bbox isn't filled in after the projection and aspect ratio are applied.

Reply:
Output looks something like this now: (code to generate this is pasted below.)
<img width="648" alt="screen shot 2018-07-09 at 9 31 59 pm" src="https://user-images.githubusercontent.com/1492803/42486023-c5b30f2c-83bf-11e8-97f7-97f5054eae6f.png">

Note that I am now creating a reusable "precip_spatial" and a non-spatial "precip" .rds file. as shown below. -- the NaNs are outside the domain of the data.

<img width="458" alt="screen shot 2018-07-09 at 9 16 37 pm" src="https://user-images.githubusercontent.com/1492803/42486031-c8d462f0-83bf-11e8-9b5a-cb007300c3b5.png">

Side now, this number of polygons renders _very_ slowly. I think we are going to want to do some resampling to a raster or visualization grid of some sort.

```r
precip_spatial <- readRDS("1_fetch/out/precip_spatial.rds")
precip <- readRDS("1_fetch/out/precip_values.rds")

view_config <- scmake("view_config")
source("1_fetch/src/map_utils.R")

bbox <- bbox_to_polygon(view_config$bbox, 
                        return_crs = view_config$projection)
county <- fetch_geoms(scmake("focus_geoms_config"), 
                      crs = view_config$projection, 
                      within = bbox)

fil <- precip$time == precip$time[1]

p <- left_join(select(precip_spatial, id), 
               select(filter(precip, fil), -time), 
               by = "id")

plot(bbox, lwd = 3, main = precip$time[1])
plot(p["precip"], border = "transparent", add = TRUE)
plot(county$geometry, add = TRUE)
plot(bbox, lwd = 3, add = TRUE)
```